public/docs/dart/latest/guide/forms.jade

```diff
@@ -0,0 +1,662 @@
+
+.callout.is-helpful
+  header NgControl requires Form
+  :marked
+    The `NgControl` is one of a family of `NgForm` directives that can only be applied to
+    a control within a `<form`> tag.
+:marked
+  Our application can ask an `NgControl` instance whether
+  the user touched the control, the value changed, or the value is valid.
+
+  `NgControl` doesn't just track state; it updates the control with special
+  Angular CSS classes, such as `ng-valid` or `ng-invalid`.
+  We can use those class names to change the appearance of the
+  control and make messages appear or disappear.
+
+  We'll explore those effects soon. Right now
+  let's **add `ng-control`to all three form controls**,
+  starting with the Name input box.
++makeExample('forms/dart/angular2_forms-6-NGCONTROL/lib/hero_form_component.html', 'ng-control-1', 'lib/hero_form_component.html')(format="linenums:6")
+:marked
+  Be sure to assign a unique name to each `ng-control` directive.
+
+.l-sub-section
+  :marked
+    Angular registers controls under their `ng-control` names
+    with the `NgForm` directive.
+    We didn't add the `NgForm` directive explicitly but it's here;
+    we'll talk it [later in this guide](#ng-form).
+
+
+.l-main-section
+  h2 Add custom CSS for visual feedback
+
+  p.
+    <code>NgControl</code> doesn't just track state. It updates the control to
+    have three classes that describe the control's state.
+
+  table
+    tr
+      th State
+      th Class if true
+      th Class if false
+    tr
+      td Control has been visited
+      td <code>ng-touched</code>
+      td <code>ng-untouched</code>
+    tr
+      td Control's value has changed
+      td <code>ng-dirty</code>
+      td <code>ng-pristine</code>
+    tr
+      td Control's value is valid
+      td <code>ng-valid</code>
+      td <code>ng-invalid</code>
+
+:marked
+  Let's add a temporary local template variable
+  <!-- TODO: make that a link to (./template-syntax.html#local-vars) -->
+  named **spy**
+  to the Name `<input>` tag and use the spy to display those classes.
+
++makeExample('forms/dart/angular2_forms-7.1-CSS-SPY/lib/hero_form_component.html', 'ng-control-2')(format="linenums:6")
+
+:marked
+  Now let's run the app, focusing our attention on the Name input box
+  and following the next four steps *precisely*:
+
+  1. Look but don't touch.
+  1. Click inside the name box, then click outside it.
+  1. Add slashes to the end of the name.
+  1. Erase the name.
+
+  We'll see the following four sets of class names and their transitions:
+figure.image-display
+  img(src="/resources/images/devguide/forms/ng-control-class-changes.png" alt="Invalid Form")
+
+:marked
+  The (`ng-valid` | `ng-invalid`) pair are the most interesting to us, because we want to send a
+  strong visual signal when the values are bad. We also want to mark required fields.
+
+  We can do both at the same time with a colored bar on the left of the input box:
+
+figure.image-display
```

Comment:
After trying this out, I think we don't need both the image and the animation, but I'm not sure which to drop. (Animation could be annoying; images take up more space.) One option is to do the following, so that people who can't see the animation/image for some reason (maybe they're blind or have processing issues) can get the info:

```
  Now run the app, and look at the Name input box.
  Follow the next four steps *precisely*:

  1. Look but don't touch.
  1. Click inside the name box, then click outside it.
  1. Add slashes to the end of the name.
  1. Erase the name.

  The classes are displayed as follows:

  1. `form-control ng-untouched ng-pristine ng-valid` (initial state)
  1. `form-control ng-pristine ng-valid ng-touched` (after clicking)
  1. `form-control ng-valid ng-touched ng-dirty` (after changing)
  1. `form-control ng-touched ng-dirty ng-invalid` (after erasing)

figure.image-display
  img(src="/resources/images/devguide/forms/control-state-transitions-anim.gif"  alt="Control State Transition")
```